lib/policial/linters/ruby.rb

```diff
@@ -32,22 +48,72 @@ def default_config_file
 
       private
 
-      def team
+      def iterate_until_no_changes(file, source)
+        processed_sources = []
+        iterations = 0
+
+        loop do
+          if processed_sources.include?(source.checksum)
+            raise InfiniteCorrectionLoop,
+                  "Detected correction loop for #{file.filename}"
+          else
+            processed_sources << source.checksum
+          end
+
+          if (iterations += 1) > RuboCop::Runner::MAX_ITERATIONS
+            raise InfiniteCorrectionLoop,
+                  "Stopping after #{iterations} iterations for #{file.filename}"
+          end
+
+          break unless (source = yield)
+        end
+      end
+
+      def build_violation(file, offense)
+        Violation.new(
+          file,
+          Range.new(offense.location.first_line, offense.location.last_line),
+          offense.message.strip,
+          offense.cop_name
+        )
+      end
+
+      def collate_corrections(file, corrector, team)
+        skips = Set.new
+        team.cops.each do |cop|
+          next if cop.corrections.empty? || skips.include?(cop.class)
+
+          cop.offenses.select(&:corrected?).each_with_index do |offense, index|
+            next unless build_violation(file, offense).on_changed_line?
+
+            corrector.corrections << cop.corrections[index]
+            skips.merge(cop.class.autocorrect_incompatible_with)
+          end
+        end
+      end
+
+      def build_team(auto_correct: false)
         cop_classes = if config['Rails']['Enabled']
                         RuboCop::Cop::Registry.new(RuboCop::Cop::Cop.all)
                       else
                         RuboCop::Cop::Cop.non_rails
                       end
 
-        RuboCop::Cop::Team.new(cop_classes, config, extra_details: true)
+        RuboCop::Cop::Team.new(
+          cop_classes,
+          config,
+          extra_details: true,
+          auto_correct: auto_correct,
+          stdin: ''
```

Comment:
Unless `stdin` is passed as an option, rubocop will try to save the modified file on disk...